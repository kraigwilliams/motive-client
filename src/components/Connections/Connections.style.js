import styled from 'styled-components'
import { colors } from '../constants'


export const ConnectionsPageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background-color: ${colors.darkgrey};
  margin: 0px;
  padding: 0;
  min-height: 100vh;
  height: 100%;
`;

export const ConnectionsHeader = styled.h1`
  font-size: 25px;
  padding-top: 60px;
  padding-bottom: 30px;
  
  color: ${colors.white};
  text-align: center;

  @media (min-width: 600px) {
    text-align: center;
    font-size: 45px;
  } 
`;

export const FriendsName = styled.h2`
  padding: 7px;
  color: ${colors.white};
  text-align: center;
  font-size: 25px;

  @media (min-width: 600px) {
    text-align: center;
    font-size: 27px;
  } 
`;

export const FriendsUserName = styled.h2`
  color: ${colors.blue};
  text-align: center;
  font-size: 20px;
  border-bottom: 1px solid ${colors.darkgrey};
  border-radius: 7px;
  line-height: 1.5;
  width: 300px;
  margin: auto;


  @media (min-width: 600px) {
    text-align: center;
  } 
`;


export const FriendsHeader = styled.h2`

  color: ${colors.white};
  text-align: center;
  line-height: 2.5;
  border-bottom: 1px solid ${colors.white};
  max-width: 300px;
  margin: auto;

  @media (min-width: 600px) {
    text-align: center;
  } 
`;

export const ConnectionsSection = styled.section`
  background-color: ${colors.darkergrey};
  text-align: center;
  padding-bottom: 70px;
  height: auto 140%;
  

  @media (min-width: 600px) {
    height: 100%;
  } 

`;

