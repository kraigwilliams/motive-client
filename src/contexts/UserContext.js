import React, { Component } from "react";
import AuthApiService from "../services/auth-api-service";
import TokenService from "../services/token-service";
import IdleService from "../services/idle-service";

const UserContext = React.createContext({
  user: {},
  error: null,
  addedConnection: null,
  setError: () => {},
  clearError: () => {},
  setUser: () => {},
  processLogin: () => {},
  processLogout: () => {},
  setAddedConnection: () => {},
});

export default UserContext;

export class UserProvider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {},
      error: null,
      addedConnection: null,
    };

    const jwtPayload = TokenService.parseAuthToken();

    if (jwtPayload) {
      this.state.user = {
        id: jwtPayload.user_id,
        firstname: jwtPayload.firstname,
        lastname: jwtPayload.lastname,
        username: jwtPayload.sub,
      };

      IdleService.setIdleCallback(this.logoutBecauseIdle);
    }
  }

  componentDidMount() {
    if (TokenService.hasAuthToken()) {
      IdleService.registerIdleTimerResets();
      TokenService.queueCallbackBeforeExpiry(() => {
        this.fetchRefreshToken();
      });
    }
  }

  componentWillUnmount() {
    IdleService.unRegisterIdleResets();
    TokenService.clearCallbackBeforeExpiry();
  }

  setError = (error) => {
    this.setState({ error });
  };

  clearError = () => {
    this.setState({ error: null });
  };

  setUser = (user) => {
    this.setState({ user });
  };

  processLogin = (authToken) => {
    TokenService.saveAuthToken(authToken);
    const jwtPayload = TokenService.parseAuthToken();
    this.setUser({
      id: jwtPayload.user_id,
      firstname: jwtPayload.firstname,
      lastname: jwtPayload.lastname,
      username: jwtPayload.sub,
    });
    IdleService.registerIdleTimerResets();
    TokenService.queueCallbackBeforeExpiry(() => {
      this.fetchRefreshToken();
    });
  };

  processLogout = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.setUser({});
  };

  logoutBecauseIdle = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.setUser({ idle: true });
  };

  fetchRefreshToken = () => {
    AuthApiService.refreshToken()
      .then((res) => {
        TokenService.saveAuthToken(res.authToken);
        TokenService.queueCallbackBeforeExpiry(() => {
          this.fetchRefreshToken();
        });
      })
      .catch((err) => {
        this.setError(err);
      });
  };

  setAddedConnection = (value) => {
    this.setState({
      addedConnection: value,
    });
  };

  render() {
    const value = {
      user: this.state.user,
      error: this.state.error,
      addedConnection: this.state.addedConnection,
      setError: this.setError,
      clearError: this.clearError,
      setUser: this.setUser,
      processLogin: this.processLogin,
      processLogout: this.processLogout,
      setAddedConnection: this.setAddedConnection,
    };
    return (
      <UserContext.Provider value={value}>
        {this.props.children}
      </UserContext.Provider>
    );
  }
}
