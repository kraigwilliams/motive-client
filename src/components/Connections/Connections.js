import React, { Component } from 'react'
import UserContext from '../../contexts/UserContext'
import ActionsService from '../../services/actions-service'
import{ ConnectionsPageWrapper, ConnectionsHeader, FriendsName, FriendsUserName, FriendsHeader, ConnectionsSection } from './Connections.style';
import { FormInput, FormWrapper, FormTitle } from '../Form/Form'
import { colors } from '../constants'
import Connection from '../Connection/Connection'


export default class Connections extends Component {
  static contextType = UserContext;

  constructor(props) {
    super(props)
    this.state = {
      query: '',
      connections: [],
      nonconnections: [],
      filteredData: [],
    }
  }

  
  handleInputChange = event => {
    const query = event.target.value;

    this.setState(prevState => {
      const filteredData = prevState.nonconnections.filter(element => {
        return element.name.toLowerCase().includes(query.toLowerCase());
      });
      return {
        query,
        filteredData
      };
    });
  };

  async getData(){
    const { user } = this.context;
    const userId = user.id;
    const connections = await ActionsService.getConnections(userId)
    if(connections) {
      this.setState({
        connections
      })
    }
    
    const nonconnections = await ActionsService.getAllNonconnections()
    if (nonconnections) {
      this.setState({
        nonconnections
      })
      const { query } = this.state;
   
      const filteredData = nonconnections.filter(element => {
       return element.name.toLowerCase().includes(query.toLowerCase());
      });
      this.setState({ 
        filteredData
      });
    }  
 }

 componentDidMount() {
   this.getData();
 }
  
  render() {
    const { connections, filteredData } = this.state;

    return(
     
      <ConnectionsPageWrapper>
          <ConnectionsHeader >
          {this.context.user.username}'s Connections
          </ConnectionsHeader>
   
          <FormWrapper backgroundcolor='none'>
            <FormTitle color={colors.coral}>
              Add Connections
            </FormTitle>

            <FormInput 
              onChange={this.handleInputChange} 
              backgroundcolor={colors.darkergrey}
              color={colors.white} 
              placeholder='Search...'
            />
            
            {/* <FormButton 
              type='submit' 
              margintop={'15px'} 
              backgroundcolor={colors.darkergrey} 
              color={colors.offwhite}
            >
              Search
            </FormButton> */}

            <FormTitle color={colors.offwhite}>
              {/* this will eventually be getting the data for search friends from DB */}
              {filteredData.map((connection, idx) => {
              return <Connection 
                firstname={connection.first_name} 
                lastname={connection.last_name}
                username={connection.username}
                id={connection.id}
              /> 
              })}

              {/* 
              <Connection /> 
              <Connection /> 
              <Connection />  
              */}
            </FormTitle>
          </FormWrapper>

          <ConnectionsSection>
            <FriendsHeader> Friends </FriendsHeader>
            {/* map through connections ins tate to render each connection detail */}
              {connections.map(friend => {
                return (
                  <>
                    <FriendsName>friend.first_name frined.last_name</FriendsName>
                    <FriendsUserName>friend.username</FriendsUserName>
                  </>
                )
              })}

            {/* hard coded for now */}
            {/* <FriendsName>Jordan Castillo</FriendsName>
            <FriendsUserName>jordanxcast</FriendsUserName>
            <FriendsName>Scott Lingner</FriendsName>
            <FriendsUserName>slingner</FriendsUserName>
            <FriendsName>Kraig Williams</FriendsName>
            <FriendsUserName>kwill</FriendsUserName> */}
      
        </ConnectionsSection>
      


        
    </ConnectionsPageWrapper >
   
    )
  }
}

