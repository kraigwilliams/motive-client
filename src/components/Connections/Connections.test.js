import React from "react";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Connections from "./Connections";
import Adapter from "enzyme-adapter-react-16";
import UserContext from "../../contexts/UserContext";

configure({ adapter: new Adapter() });
describe(`<Connections />`, () => {
  //Snapshot Testing
  it("renders Connections by default", () => {
    const wrapper = shallow(
      <UserContext.Provider>
        <Connections />
      </UserContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
