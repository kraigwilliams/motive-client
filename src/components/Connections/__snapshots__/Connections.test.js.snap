// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Connections /> renders Connections by default 1`] = `<Connections />`;
